sys/arduino/SPI.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2019 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     sys_arduino_api
+ * @{
+ *
+ * @file
+ * @brief       Implementation of the Arduino 'SPI' interface
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ *
+ * @}
+ */
+
+#ifdef MODULE_PERIPH_SPI
+
+extern "C" {
+#include "assert.h"
+}
+
+#include "SPI.h"
+
+SPISettings::SPISettings(uint32_t clock_hz, uint8_t bitOrder, uint8_t dataMode)
+{
+    static const spi_clk_t clocks[] = {
+        SPI_CLK_10MHZ, SPI_CLK_5MHZ, SPI_CLK_1MHZ, SPI_CLK_400KHZ
+    };
+    static const uint32_t steps [] = {
+        1000000, 5000000, 1000000, 400000
+    };
+
+    assert(bitOrder == MSBFIRST);
+    switch(dataMode) {
+        default:
```

Comment:
The order of cases as no syntactic meaning. Also removing the `case SPI_MODE0:` after the `default:` would not be an syntactic change.

Having the `SPI_MODE0` as first entry makes the cases alphabetically sorted, so easier to read. And mode 0 is the most common, so using this as a fall bock in case of an invalid `dataMode` seems to be a sane fallback.

Reply:
> And mode 0 is the most common, so using this as a fall bock in case of an invalid `dataMode` seems to be a sane fallback.

That's clear.

Another possibility having the same effect of readability would have been to use the reverse order including label `SPI_MODE0`. But it's probably as often a matter of taste. I've never seen it like this before. I've always seen `default` as the last case, probably to make it clear that it's the fallback if no other case applies. But, I'm fine with it.